### PR TITLE
Sync tabs with swipe view in FeatureForm.

### LINKED ISSUE #5242: Tabs are not following the selected container.
Hello,

The tabs are not adjusting with the selected container.

I believe that moving the containers to the right or left within the form should also shift the tabs accordingly.

Currently, when I move to the left to access the containers on the right, the tabs do not move, causing some tab aliases to be hidden.

While this is not a critical issue (no critical issues remain for me on the QField app), it affects the overall quality of the application.

https://github.com/opengisch/QField/assets/72577399/919a8495-09f2-4f02-86ea-07e877fd58ed

Thank you.

Desktop
OS: [Windows 10]
QGIS Version [3.36.1]
QFieldSync Version [4.9.1]

Mobile
Device: [Samsung Galaxy S9]
OS: [Android 11]
QField version: [master-dev (14a0be)]

### Additional information
* Problem started happening recently, didn't happen in an older version of QField: [No sure]
* Problem can be reliably reproduced, doesn't happen randomly: [Yes]
* Problem happens with all files and projects, not only some files or projects: [Yes]

### PR DESCRIPTION
**Pull request by mohsenD98**
In this PR, we address two minor issues:

1- Fix #5242: The issue where tabs are not following the selected container has been resolved.
2- Additionally, we've fixed the elide problem in MapValue.

These changes aim to improve the overall user experience and stability of application.

**Resolved video:**   
  
[Screencast from 2024-05-31 07-34-16.webm](https://github.com/opengisch/QField/assets/36326627/dbb7bc20-ef10-4561-acd8-d54109207912)

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for ca96e98f3db3f94c4d11df3a77f17f8a1531f80e: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5281-56e0c44997e65d26d3f901e535abc781b14e972d-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5281-56e0c44997e65d26d3f901e535abc781b14e972d-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5281-56e0c44997e65d26d3f901e535abc781b14e972d-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5281-56e0c44997e65d26d3f901e535abc781b14e972d-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5281-56e0c44997e65d26d3f901e535abc781b14e972d-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5281-56e0c44997e65d26d3f901e535abc781b14e972d-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5281-56e0c44997e65d26d3f901e535abc781b14e972d-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5281-56e0c44997e65d26d3f901e535abc781b14e972d-android-x86.apk)
</details>

File: src/qml/FeatureForm.qml [hunk 1]
```diff
@@ -76,86 +76,92 @@ Page {
...
+      highlightFollowsCurrentItem: true
+      highlight: Item {
+        Rectangle {
+          height: 2
+          color: Theme.mainColor
+          radius: 4
+          width: parent.width
+          anchors.bottom: parent.bottom
+        }
+      }
```
Comment (nirvn) [inline src/qml/FeatureForm.qml:95]: If we go for this fix, we will need to implement it as a QfTabBar item, and replace our uses of TabBar {} with that across the board. If we don't do that, we'll end up with inconsistent tab bars.

File: src/qml/FeatureForm.qml [hunk 2]
```diff
@@ -76,86 +76,92 @@ Page {
...
+
     /**
      * The main form content area
      */
     SwipeView {
       id: swipeView
       Layout.fillWidth: true
       Layout.fillHeight: true
-      currentIndex: tabRow.currentIndex
+      currentIndex: tabBarListView.currentIndex
```
Comment (nirvn) [inline src/qml/FeatureForm.qml:164]: For the sake of avoiding noise, I would actually keep the same id, i.e. tabRow.

File: src/qml/editorwidgets/ValueMap.qml [hunk 3]
```diff
@@ -135,6 +135,7 @@ EditorWidgetBase {
 
             Text {
               id: innerText
+              width: Math.min(flow.width - 32 , implicitWidth)
```
Comment (nirvn) [inline src/qml/editorwidgets/ValueMap.qml:138]: :+1:

**Review by nirvn — APPROVED** [review]
Can't spot a regression, and it does fix the issue you were trying to address. All good.

**Pull request merged**